invenio_app_ils/circulation/api.py

```diff
@@ -104,6 +104,9 @@ def request_loan(
     **kwargs
 ):
     """Create a new loan and trigger the first transition to PENDING."""
+    if current_user.is_anonymous:
```

Comment:
After the chat we had today, I realized that I have asked you to do this modification but it is not right :( I am really sorry about that.

To explain, here the 2 cases:
- this method is called in a HTTP request: the view is protected by `needs_permission` so the `current_user` is set, no need the extra check here
- this method is called via command line (if we are for example operating in case of user support) then we need to pass the param `transaction_user_pid`, it is ok to fail otherwise.
If we keep the if here, we won't be able to call this method when we are not in a request context, via command line.

Really sorry... I will point out all the others to remove.